### PR TITLE
Milestone4

### PR DESCRIPTION
**Pull request by saimaAnis21**
In this project we did:
1- Logic folder was added
2- calculate.js and operate.js modules were added to the folder
3- calculate.js was added to the App component

### CODE REVIEW

File: src/logic/calculate.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+function calculate(calculator, buttonName) {
+  let total = 0;
+  let next = 0;
+  if (buttonName === '+/-') {
+    total = calculator.total * -1;
+    next = calculator.next * -1;
+  } else if (buttonName === '+' || buttonName === '-' || buttonName === 'X' || buttonName === '%') {
+    operate(total, next, buttonName);
+  }
+}
```
Comment (Berabjesus) [inline src/logic/calculate.js]: - [ ] Have the `calculate` function return the mutated final total value.

File: src/logic/calculate.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import operate from './operate';
+
+function calculate(calculator, buttonName) {
+  let total = 0;
+  let next = 0;
+  if (buttonName === '+/-') {
+    total = calculator.total * -1;
+    next = calculator.next * -1;
+  } else if (buttonName === '+' || buttonName === '-' || buttonName === 'X' || buttonName === '%') {
+    operate(total, next, buttonName);
```
Comment (Berabjesus) [inline src/logic/calculate.js]: - [ ] The operate function is called but the return value of the `operate` function is not casted into any variable. Please cast the returned value into the appropriate variable.
```
total =  operate(total, next, buttonName)
```

**Review by Berabjesus — CHANGES_REQUESTED** [review]
# Status: CHANGES REQUIRED :warning: :construction: 
Hey @saimaAnis21 👋 
Great start to this milestone. You have implemented most of the specifications required.
### However, you do have a few changes to carry out before we proceed so please take a look at the things you have to work on and their respective review comments below.

- [ ] `calculate.js` uses `button name` argument to mutate `calculator data object` and return its modified version. For instance if button's name == "+/-", it should multiply total and next by -1 to make it negative or positive. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone4/calculate-mutates-object.png)).
- Currently, your `calculate.js` returns nothing but it should mutate the data and return the total value.

- [ ] `calculate.js` should have some sort of condition to check if the button name is `AC` and handle the corresponding action.
- [ ] `operate.js` has an `operate` function that switches `operation` and uses `plus`, `minus`, `times`, `div` and `modifier` (turning the number into a percentage based number, `e.g. 95 % == .95`) functions ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone4/performOperate.png)). Currently, you have only done that for the `+` operator.
- [ ] [OPTIONAL] In `calculate.js` use ES6 syntax to destructure the prop like this `let { total, next, operation } = dataset;` 

#### **If you have any questions, let me know by leaving a comment here and mentioning me. I will be happy to help** 
**HAPPY CODING :computer:[Bereket Beshane](https://bereketababu.me)** | [LinkedIn](https://linkedin.com/in/bereketbeshane) | [GitHub](https://github.com/Berabjesus)

File: src/logic/calculate.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import operate from './operate';
+
+function calculate(calculator) {
+  let { total, next } = calculator;
```
Comment (Forison) [inline src/logic/calculate.js]: * [ ] Kindly revert to the old code. The function was not returning anything. all the previous reviewer sought to say is for you to return the mutated calculator object

- `calculate` accepts `calculator data object` and a `button name` as arguments ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone4/buttonAsArgument.png)) 
The calculator data object is going to handle the `mutation` of the current state of the application.
- `calculator data object` contains three properties: `total`, `next` and `operation` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone4/totalNextOperate.png))
This implies  that our function could be defined like this ` const calculate ({total, next, operator}, buttonName)`

- uses `button name` argument to mutate `calculator data object` and return its modified version. For instance if button's name == "+/-", it should multiply total and next by -1 to make it negative or positive. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone4/calculate-mutates-object.png))

Have conditional statement that mutate the calculator data `{total, next, operator}` object based on the value of the second argument `buttonName`
button names like `A/C`, `+/-`, `%`  etc... mutate the state.
**Scenario**
> When user clicks A/C user wants to reset the calculator, this translate to setting attributes of data object to null or empty... 

> When user clicks `+/-` user wants negate operands, hence we mutate the state as such. 
....
> When All the necessary mutations are done, return the current data object.

File: src/logic/operate.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+      result = nmbrOne - nmbrTwo;
+      break;
+    case 'X':
+      result = nmbrOne * nmbrTwo;
+      break;
+    case '%':
+      result = nmbrOne / 100;
+      break;
+    case '&#247;':
+      result = nmbrOne / nmbrTwo;
```
Comment (Forison) [inline src/logic/operate.js]: * [ ] Let us handle the division by zero, in this case, 
We can do it this way
`result = nmbrTwo === 0 ? 'undefined' : nmbrOne / nmbrTwo;`

**Review by Forison — CHANGES_REQUESTED** [review]
# Status::Change Required❌

### hello @saimaAnis21 

### Good job on addressing the issue raised in the previous review, I think you misunderstood some of the recommendations, kindly check my comments below😇.

That is it kindly make the change and submit a re-review

#### Happy coding
Addo Forison

**Review by Forison — APPROVED** [review]
# Status::Approved✔️

#### Good job on addressing the issues raised, `operate` and `calculate` look good, kindly go ahead and merge the branch and proceed to the next milestone🤗.

#### Happy coding👩‍💻

Addo Forison

**Pull request merged**